### ISSUE TITLE
small bug in ocl_gpu_kernel.cpp

### ISSUE THREAD

**Issue by sclavel** [oneapi-src/oneDNN#721]
in the file /src/gpu/ocl/ocl_gpu_kernel.cpp, the call to clCreateProgramWithBinary() passes `sizeof(size_t)` as the second argument, while it should be the number of devices (as per openCL documentation: `cl_uint num_devices`), so it should be replaced by `1`.

```diff
@@ -93,7 +93,7 @@ status_t ocl_gpu_kernel_t::realize(
     assert(binary_size > 0);
 
     auto program = clCreateProgramWithBinary(compute_engine->context(),
-            sizeof(size_t), &dev, &binary_size, &binary_buffer, nullptr, &err);
+            1, &dev, &binary_size, &binary_buffer, nullptr, &err);
     OCL_CHECK(err);
     err = clBuildProgram(program, 1, &dev, nullptr, nullptr, nullptr);
     OCL_CHECK(err);
```

Some openCL drivers might silently ignore this error, but we saw some drivers that failed with the incorrect argument.

**Comment by dyoussif** [comment]
Hi @sclavel, thanks for pointing this out. This will be addressed.

**Comment by vpirogov** [comment]
Resolved with a593999abfcb5aa617c63b3c6ae9762c4efb8aa3.
